test/integration/src/test/java/io/pravega/test/integration/KVTableWithStreamTest.java

```diff
@@ -0,0 +1,769 @@
+
+        final ScalingPolicy scalingPolicy = ScalingPolicy.fixed(5);
+        final StreamConfiguration strmConfig = StreamConfiguration.builder()
+                .scalingPolicy(scalingPolicy)
+                .build();
+
+        // Creating Stream1 under Scope1.
+        assertTrue("The requested Stream must get created", this.controller.createStream(newScope1,
+                streamName1, strmConfig).join());
+
+        val strmSegments1 = this.controller.getCurrentSegments(newScope1, streamName1).join();
+        for (val s : strmSegments1.getSegments()) {
+            // We know there's nothing in these segments. But if the segments hadn't been created, then this will throw
+            // an exception.
+            this.streamSegmentStore.getStreamSegmentInfo(s.getScopedName(), TIMEOUT).join();
+        }
+
+        // KVTable and its table segments should get deleted but Stream and its stream segments should exist and +
+        // we should be able to read/write from them.
+        ResultObserver<DeleteKVTableStatus> result3 = new ResultObserver<>();
+        this.controllerService.deleteKeyValueTable(ModelHelper.createKeyValueTableInfo(newScope1,
+                newKVT1.getKeyValueTableName()), result3);
+        assertEquals("The requested table to delete must not be found for invalid Scope name",
+                DeleteKVTableStatus.Status.SUCCESS, result3.get().getStatus());
+
+        for (val s : tableSegments1.getSegments()) {
+            AssertExtensions.assertSuppliedFutureThrows(
+                    "Segment " + s + " has not been deleted.",
+                    () -> this.tableStore.get(s.getKVTScopedName(), Collections.singletonList(new
+                            ByteArraySegment(new byte[1])), TIMEOUT),
+                    ex -> ex instanceof StreamSegmentNotExistsException);
+        }
+
+        for (val s : strmSegments1.getSegments()) {
+            // We know there's nothing in these segments. But if the segments hadn't been created, then this will throw
+            // an exception.
+            this.streamSegmentStore.getStreamSegmentInfo(s.getScopedName(), TIMEOUT).join();
+        }
+
+        try (ConnectionFactory connectionFactory = new SocketConnectionFactoryImpl(ClientConfig.builder().build());
+             ClientFactoryImpl clientFactory = new ClientFactoryImpl(newScope1, this.controller, connectionFactory);
+             ReaderGroupManager readerGroupManager = new ReaderGroupManagerImpl(newScope1, this.controller, clientFactory)) {
+
+            log.info("Creating writer");
+            @Cleanup
+            final EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName1,
+                    new JavaSerializer<>(), EventWriterConfig.builder().build());
+
+            // start writing events to the stream.
+            for (int i = 0; i < numEvents; i++) {
+                log.info("Writing event {}", streamEvent);
+                writer.writeEvent(routingKey, streamEvent);
```

Comment:
async

Reply:
Its just 1 writer appending some 100 events in 1 stream, do you suggest here to make this operation async ?

Reply:
No. I am saying that `writer.writeEvent` is an async call. It returns a `CompletableFuture`. This indicates that the event itself is not going to be (guaranteed to be) written until that Future is completed. In your code, you are just initiating the Event write but never waiting for it so there's no guarantee as to what gets written - if anything. Also you will not be notified of any errors during the write (since errors will complete the returned Future in an exceptional way).

Please go through your entire PR and look for any APIs that you invoke which return a `CompletableFuture`. Unless you have a very good reason (in which case please document using a comment), you need to wait on that future to complete, otherwise your test may not be testing anything after all.

This is the reason why pointed out several places in this PR where you are not waiting for returned futures to complete.

Reply:
okay, got it. I'll complete all the `CompletableFuture` API calls wherever used to make them work correctly as suggested. Thanks @andreipaduroiu